rezz.lic

```diff
@@ -1,118 +1,131 @@
-#   Documentation: https://elanthipedia.play.net/Lich_script_repository#rezz
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#rezz
+  
+  Define a waggle sets for rezz and rejuv if you don't want to use the default values in base-spells.yaml
+  
+  EXAMPLE
+  waggle_sets:
+    rezz:
+      Resurrection:
+        mana: 17
+    rejuv:
+      Rejuvenation:
+        mana: 10
+        cambrinth:
+        - 15
+
+  In order to infuse directly from your mana pool add the following to your yaml:
+  osrel_no_harness: true
+  
+=end
 
 custom_require.call(%w[common common-arcana events drinfomon spellmonitor])
 
-# rezz waggle set with POM if you want
-
 class Rezz
   include DRC
   include DRCA
-
+  
   def initialize
+    DRStats.cleric? ? nil : begin echo '*** Nice try, heathen! ***'; exit end
+	
     arg_definitions = [
       [
         { name: 'person', regex: /\w+/, description: 'Person to rezz' }
       ]
     ]
-
-    @rejuv_mana = 0
+	
     args = parse_args(arg_definitions)
     @settings = get_settings
-    rezz_prep(args.person)
+    
+    @person = args.person.capitalize
+	/the body of (#{@person}.*) who/ =~ bput('look people', /the body of (#{@person}.*) who/)
+	@person = Regexp.last_match(1)
+    
+    @rezz = if @settings.waggle_sets['rezz']
+               @settings.waggle_sets['rezz'].values.first
+           else
+	         get_data('spells').spell_data['Resurrection']
+	       end
+    	   
+    @rejuv = if @settings.waggle_sets['rejuv']
+               @settings.waggle_sets['rejuv'].values.first
+           else
+	         get_data('spells').spell_data['Rejuvenation']
+	       end
+
+    @rejuv['cast'] = "cast #{@person}"
+	
+	check_favor
+    rejuv
+    cast_rezz
+    find_soul
+    soul_bond
+    rejuv(true)
+    gesture
   end
-
-  def rezz_prep(person)
-    # cast buffs if set
-    check_buffs
-
-    # perceive body & rejuv
-    rejuv(person)
-
-    # cast rezz
-    mana = cast_rezz
-
-    # infuse untill see spirit
-    find_soul(person, mana)
-
-    # snap cast soul bonding
-    cast_soulbonding(person)
-
-    # check for silver again. recast rejuv (or infuse again) if not
-    rejuv(person)
-
-    # gesture
-    bput("gest #{person}", 'Roundtime')
+  
+  def check_favor
```

Comment:
When I first added that, it was so I could have the script echo some message about if the person had low favors, but I didn't do that in the end.  I just have it so it exits the script if the person has no favors, rather than preparing the body fully and then finding out when trying to gesture.

Now that I think about it though, it should rejuv and then exit if they have no favors.